src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -89,15 +91,26 @@ public RoomDto startGame(String roomId, StartGameRequest request) {
     }
 
     // Initialize and add a game object from a room object
-    public void createAddGameFromRoom(Room room) {
+    public Game createAddGameFromRoom(Room room) {
         Game game = GameMapper.fromRoom(room);
         currentGameMap.put(room.getRoomId(), game);
+        return game;
     }
 
-    // Test the submission and return a socket update.
-    public GameDto testSubmission(String userId, String roomId) {
-        // TODO: Get the player and game.
-        return submitService.testSubmission(new Player(), new Problem());
+    // Test the submission, return the results, and send a socket update
+    public SubmissionDto submitSolution(String roomId, SubmissionRequest request) {
+        Game game = currentGameMap.get(roomId);
+
+        if (request.getInitiator() == null || request.getCode() == null || request.getLanguage() == null) {
+            throw new ApiException(GameError.EMPTY_FIELD);
+        }
+
+        String initiatorUserId = request.getInitiator().getUserId();
+        if (!game.getPlayers().containsKey(initiatorUserId)) {
+            throw new ApiException(GameError.INVALID_PERMISSIONS);
+        }
+
+        return submitService.submitSolution(game, request);
     }
```

Comment:
Do we need this method? Could we just directly use the `SubmitService.java`'s `submitSolution` method, and autowire the `GameManagementService` in order to get the correct `Game`? That might get rid of what we had predicted to be the purpose of this service class, so I'm open to thoughts.

Reply:
Kinda like the inverted design I think right (was a very long time ago so I don't recall all the details lol) as opposed to the alternative design? I forget what we ended up choosing but if we want to re-decide on that during standup I can go ahead and make some of those changes

Reply:
Ok I think that actually works, I had just kinda forgotten that discussion and hadn't seen it in action. I think this design should be fine for this 👍